app/src/main/java/com/sthoray/allright/data/api/AllGoodsApi.kt

```diff
@@ -31,4 +32,14 @@ interface AllGoodsApi {
      */
     @POST("search")
     suspend fun searchForListings(@Body searchRequest: SearchRequest = SearchRequest()): Response<SearchResponse>
+
+    /**
+     * Get listing of that id.
+     *
+     * @param listingId id of the product.
+     * @return a [Listing] response.
+     */
+    @GET("product/{productId}")
+    suspend fun getListing(listingId: Int) : Response<Listing>
```

Comment:
The `listingId` parameter will need a path annotation before it, i.e.`@Path("productId")`